src/MakeJsonCommand.php

```diff
@@ -48,6 +48,9 @@ class MakeJsonCommand extends WP_CLI_Command {
 	 * [--pretty-print]
 	 * : Pretty-print resulting JSON files.
 	 *
+	 * [--use-map[=<paths>]]
+	 * : Whether to use a mapping file for the strings. Separate with `,` to specify multiple. Defaults to `build/map.json`.
```

Comment:
I don't think we should default to any value here. This makes an assumption about the directory structure that won't fit for many projects.

Relatedly, would it perhaps make sense to allow JSON input such as `--input="{ \"src/index.js\": \"build/index.js\" }"` so one doesn't have to _have_ a map file?

Reply:
Hmm, nice suggestion, but my problem is now how do I parse the argument? The first thing I thought of was just to use `json_decode`, but that means the user has to put quotes around the strings, are we fine with that?

Reply:
Actually, nevermind, what do you think of this regex:

```regex
(?:^|(?<=,))\s*([^{[}\]"',]*?)\s*(?:$|(?=,))
```

We could maybe use that to add quotes around the strings

Reply:
`Utils\parse_shell_arrays()` already does most of the heavy lifting here. It calls `Utils\is_json()` and runs `json_decode()` if needed. So we don't need to do anything extra here.

Check out `MakePotCommand` and how the `headers` argument (which allows JSON) is parsed there.

Reply:
Thank you, I didn't know of that, I'll look into that!